src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -52,22 +65,101 @@ def validate_client_parameters(namespace):
         else:
             raise ValueError("Storage account '{}' not found.".format(n.account_name))
 
-def get_content_setting_validator(settings_class):
+def validate_source_uri(namespace):
+    usage_string = 'invalid usage: supply only one of the following argument sets:' + \
+                   '\n\t   --source-uri' + \
+                   '\n\tOR --source-container --source-blob [--source-snapshot] [--source-sas]' + \
+                   '\n\tOR --source-share --source-path [--source-sas]'
+    ns = vars(namespace)
+    validate_client_parameters(namespace) # must run first to resolve storage account
+    storage_acc = ns.get('account_name', None) or os.environ.get('AZURE_STORAGE_ACCOUNT')
+    uri = ns.get('copy_source', None)
+    container = ns.pop('source_container', None)
```

Comment:
pop = things that need to be removed from the namespace
get = things that need to stay

The alternative was having a bunch of obnoxious "del namespace.blah" lines at the end. This was a more concise approach.

---
In reply to: [77393712](https://github.com/Azure/azure-cli/pull/804#discussion_r77393712) [](ancestors = 77393712)